### PR TITLE
rework syscall definitions in linux

### PR DESCRIPTION
**Pull request by ArsenArsen**
potential todo: add a CI pipeline to update this monthly

### CODE REVIEW

File: sysdeps/linux/include-internal/linux/unistd.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#include <syscallnos.h>
```
Comment (64) [inline sysdeps/linux/include-internal/linux/unistd.h:1]: Can't this live in `include/bits`?

File: sysdeps/linux/update-syscall-list.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+logging.basicConfig(level=logging.DEBUG)
+LOGGER = logging.getLogger(__name__)
+
+# list of architecture directories to populate with syscall numbers
+WANTED_ARCHES = ["riscv64", "x86_64"]
+
+argp = argparse.ArgumentParser()
+argp.add_argument("kver", help="Kernel version to download", default="latest")
+
+GENERATE_ARCHES = ["x86_64", "riscv64"]
```
Comment (64) [inline sysdeps/linux/update-syscall-list.py:29]: This appears to be unused

**Review by 64 — CHANGES_REQUESTED** [review]
Might have some more comment later but the general approach LGTM. What I'd say is that it'd be nice to sort the generated files somehow (either alphabetically or by syscall number) so that they're easier to search + order won't randomly change when the headers are pulled again

File: sysdeps/linux/include-internal/linux/unistd.h (hunk 1 above)
Comment (ArsenArsen) [reply sysdeps/linux/include-internal/linux/unistd.h:1]: it's not installed

Comment (64) [reply sysdeps/linux/include-internal/linux/unistd.h:1]: Can't we just not install it then? bc having both `include/bits` and `include-internal` seems a bit unnecessary to me

**Review by 64 — APPROVED** [review]
LGTM, wouldn't mind if the header was moved still but I'm not too fussed about it

**Review by Geertiebear — APPROVED** [review]
LGTM, does the script need to be run by users ever or is it only used for development?

**Comment by ArsenArsen** [comment]
not even devs would need to run it normally

**Pull request merged**